### ISSUE TITLE
Multiple instances of Update.exe causes updates to get stuck.

### ISSUE THREAD

**Issue by mossyhub** [velopack/velopack#364]
### OS

Windows (all)

### Programming Language

dotnet 6

### VPK Version

vpk v0.0.869

### Library Version

Nuget v0.0.869

### What happened?

If multiple instances of Update.exe get launched when an update is already in progress, the update never finishes.

While this is certainly a very corner case, and only caused by my specific method of launching my application ( externally calling "Update.exe start myproccess.exe" via a scheduled task that repeats every 1 minute  ). I wont dive into why I need to be doing this, however, I think it wouldn't be a bad idea, even outside of my case, to add logic in Update.exe to check if an instance is already running and kill itself/the new instance if there is.

### Relevant log output

```shell
[update:10208] [17:40:31] [INFO] --
[update:10208] [17:40:31] [INFO] Starting Velopack Updater (0.0.869)
[update:10208] [17:40:31] [INFO]     Location: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\Update.exe
[update:10208] [17:40:31] [INFO]     CWD: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp
[update:10208] [17:40:31] [INFO]     Verbose: false
[update:10208] [17:40:31] [INFO]     Silent: false
[update:10208] [17:40:31] [INFO]     Log File: None
[update:10208] [17:40:31] [INFO] Command: Start
[update:10208] [17:40:31] [INFO]     Wait: WaitParent
[update:10208] [17:40:31] [INFO]     Exe Name: Some("MyApp.exe")
[update:10208] [17:40:31] [INFO]     Exe Args: None
[update:10208] [17:40:31] [INFO] Reading parent process information.
[update:10208] [17:40:31] [INFO] Waiting 60000ms for parent process (9592) to exit.
[update:10208] [17:40:34] [WARN] Failed to locate app manifest (This application is missing a package manifest (.nuspec) or it could not be parsed.), trying legacy package resolution...
[update:10208] [17:40:34] [INFO] Attempting to auto-detect package in: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\packages
[update:12444] [17:40:54] [INFO] --
[update:12444] [17:40:54] [INFO] Starting Velopack Updater (0.0.869)
[update:12444] [17:40:54] [INFO]     Location: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\update.exe
[update:12444] [17:40:54] [INFO]     CWD: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp
[update:12444] [17:40:54] [INFO]     Verbose: false
[update:12444] [17:40:54] [INFO]     Silent: false
[update:12444] [17:40:54] [INFO]     Log File: None
[update:12444] [17:40:54] [INFO] Command: Start
[update:12444] [17:40:54] [INFO]     Wait: NoWait
[update:12444] [17:40:54] [INFO]     Exe Name: Some("MyApp.exe")
[update:12444] [17:40:54] [INFO]     Exe Args: None
[update:12444] [17:40:54] [WARN] Failed to locate app manifest (This application is missing a package manifest (.nuspec) or it could not be parsed.), trying legacy package resolution...
[update:12444] [17:40:54] [INFO] Attempting to auto-detect package in: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\packages
[update:9672] [17:41:54] [INFO] --
[update:9672] [17:41:54] [INFO] Starting Velopack Updater (0.0.869)
[update:9672] [17:41:54] [INFO]     Location: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\update.exe
[update:9672] [17:41:54] [INFO]     CWD: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp
[update:9672] [17:41:54] [INFO]     Verbose: false
[update:9672] [17:41:54] [INFO]     Silent: false
[update:9672] [17:41:54] [INFO]     Log File: None
[update:9672] [17:41:54] [INFO] Command: Start
[update:9672] [17:41:54] [INFO]     Wait: NoWait
[update:9672] [17:41:54] [INFO]     Exe Name: Some("MyApp.exe")
[update:9672] [17:41:54] [INFO]     Exe Args: None
[update:9672] [17:41:54] [WARN] Failed to locate app manifest (This application is missing a package manifest (.nuspec) or it could not be parsed.), trying legacy package resolution...
[update:9672] [17:41:54] [INFO] Attempting to auto-detect package in: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\packages
[update:9992] [17:42:54] [INFO] --
[update:9992] [17:42:54] [INFO] Starting Velopack Updater (0.0.869)
[update:9992] [17:42:54] [INFO]     Location: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\update.exe
[update:9992] [17:42:54] [INFO]     CWD: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp
[update:9992] [17:42:54] [INFO]     Verbose: false
[update:9992] [17:42:54] [INFO]     Silent: false
[update:9992] [17:42:54] [INFO]     Log File: None
[update:9992] [17:42:54] [INFO] Command: Start
[update:9992] [17:42:54] [INFO]     Wait: NoWait
[update:9992] [17:42:54] [INFO]     Exe Name: Some("MyApp.exe")
[update:9992] [17:42:54] [INFO]     Exe Args: None
[update:9992] [17:42:54] [WARN] Failed to locate app manifest (This application is missing a package manifest (.nuspec) or it could not be parsed.), trying legacy package resolution...
[update:9992] [17:42:54] [INFO] Attempting to auto-detect package in: C:\WINDOWS\ServiceProfiles\LocalService\AppData\Local\MyApp\packages
[update:10108] [17:43:54] [INFO] --
```

**Comment by caesay** [comment]
That log doesn't tell me anything unfortunately (it just looks like your app is broken, because it can't find an application manifest). Probably if something went wrong, it would have been earlier in the log. Which if you're running your app every minute, I presume that earlier log is long gone.

I have added a lockfile preventing simultaneously running Velopack operations in the latest release, please update Velopack to the latest version and confirm if the issue persists. If it does, please open a new issue!

**Comment by mossyhub** [comment]
Awesome, i will give that new version a try.

As for my app being broken: In my Azure build pipelines I am signing my nupkg files (and therefore also manually managing my manifest json files) and have created my own custom UpdateManager class to handle downloading and validating that they are signed and pass the nuget integrity check before placing them in the expected final package location. I think my signing of the nupkg files is causing update.exe to fail to automatically parse the nuspec file, so it tries other methods but seems to take longer than 1 minute to do that....so in that space of time I have already launched another update.exe. I have already modified my launcher task (which uses powershell.exe -command "...\update.exe ...") to first make sure my app, nor update.exe is running before running "update.exe start" again.
